### PR TITLE
Auto close extensions in SQL domains

### PR DESCRIPTION
**Pull request by proteusvacuum**
http://manage.dimagi.com/default.asp?228836#1159086

@czue 
buddy: @emord

### CODE REVIEW

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 1]
```diff
@@ -429,6 +429,20 @@ def get_reverse_indexed_cases(domain, case_ids):
...
     @staticmethod
+    def get_extension_chain(domain, case_ids):
+        assert isinstance(case_ids, list)
+
+        extension_chain_ids = set()
+        incoming_extensions = set(CaseAccessorSQL.get_extension_case_ids(domain, case_ids))
+        all_extension_ids = set(incoming_extensions)
+        new_extensions = set(incoming_extensions)
+        while new_extensions:
+            new_extensions = set(CaseAccessorSQL.get_extension_case_ids(domain, list(new_extensions)))
```
Comment (czue) [inline corehq/form_processor/backends/sql/dbaccessors.py]: is there any concern about circular references here causing an infinite loop? wondering if it makes sense to subtract `all_extension_ids` from this

**Comment by czue** [comment]
looks great - just had one question.

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 2]
```diff
@@ -429,6 +429,20 @@ def get_reverse_indexed_cases(domain, case_ids):
...
+            all_extension_ids = all_extension_ids | new_extensions
```
Comment (snopoke) [inline corehq/form_processor/backends/sql/dbaccessors.py]: What's the difference between `all_extension_ids` and `extension_chain_ids`? I don't see the need for both.

Comment (proteusvacuum) [inline corehq/form_processor/backends/sql/dbaccessors.py]: Updated

File: corehq/form_processor/backends/sql/dbaccessors.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/form_processor/backends/sql/dbaccessors.py]: Good catch. Fixed.

**Pull request merged**